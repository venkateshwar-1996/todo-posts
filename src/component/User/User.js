import React from "react";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";
import PropTypes from "prop-types";

const User = ({ username, name, email, website, company }) => {
  return (
    <Row data-testid={'user'}>
      <Col md={6} xs={12} className="card mb-4 bg-white text-dark post">
        <p>User Name: {username}</p>
        <p>Name: {name}</p>
        <p>Email: {email}</p>
        <p>Website: {website}</p>
      </Col>
      <Col md={6} xs={12} className="card mb-4 bg-dark text-white post">
        <p>Work: {(company || {}).name}</p>
        <p>Slogan: {(company || {}).catchPhrase}</p>
        <p>Base: {(company || {}).bs}</p>
      </Col>
    </Row>
  );
};

User.propTypes = {
  username: PropTypes.string,
  name: PropTypes.string,
  email: PropTypes.string,
  website: PropTypes.string,
  company: PropTypes.object,
};
export default User;
