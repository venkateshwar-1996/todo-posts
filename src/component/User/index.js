import User from "./User";
import withLoadingScreen from '../../HOC/withLoadingScreen';
import { MyLoader } from "../../helpers/loader";

export default withLoadingScreen(User, MyLoader);