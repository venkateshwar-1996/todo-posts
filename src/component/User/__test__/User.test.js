import React from "react";
import ReactDOM from "react-dom";
import User from "../User";
import { MemoryRouter } from "react-router-dom";
import { render } from "@testing-library/react";


it("Render without crashing", () => {
  const div = document.createElement("div");
  const company = {
    name: "jmr",
    catchPhrase: "nostalgia",
    bs: "Food"
  };

  ReactDOM.render(
    <MemoryRouter>
      <User
        username={"v@fegsbdf"}
        name={"Venkatesh"}
        email={"venkateshwar1996@gmail.com"}
        website={"google.com"}
        company={company}
      />
    </MemoryRouter>,
    div
  );
});

it("Render User correctly", () => {
  const { getByTestId } = render(
    <MemoryRouter>
      <User
        username={"v@fegsbdf"}
        name={"Venkatesh"}
        email={"venkateshwar1996@gmail.com"}
        website={"google.com"}
        company={{}}
      />
    </MemoryRouter>
  );
  expect(getByTestId("user")).toMatchSnapshot();
});
