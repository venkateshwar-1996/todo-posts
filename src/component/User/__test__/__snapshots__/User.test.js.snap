// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render User correctly 1`] = `
<div
  class="mt-4 row"
  data-testid="user"
>
  <div
    class="card mb-4 bg-dark text-white post col-md-6 col-12"
  >
    <p>
      User Name: 
      v@fegsbdf
    </p>
    <p>
      Name: 
      Venkatesh
    </p>
    <p>
      Email: 
      venkateshwar1996@gmail.com
    </p>
    <p>
      Website: 
      google.com
    </p>
  </div>
  <div
    class="card mb-4 bg-white text-dark post col-md-6 col-12"
  >
    <p>
      Work: 
    </p>
    <p>
      Slogan: 
    </p>
    <p>
      Base: 
    </p>
  </div>
</div>
`;
